tutorials/xarray_tutorial.py

```diff
@@ -6,295 +6,344 @@
+# Other useful unit integration features include:
 #
-# As an example, we calculate geostropic wind at 500 hPa below:
+# Unit-based selection/indexing:
+
+heights_at_45_north = data['Geopotential_height_isobaric'].metpy.sel(
+    latitude=45 * units.degrees_north,
+    vertical=300 * units.hPa
+)
+heights_at_45_north
 
-lat, lon = xr.broadcast(y, x)
-dx, dy = mpcalc.lat_lon_grid_deltas(lon, lat, geod=pyproj_crs.get_geod())
-heights = data['height'].metpy.loc[{'time': time[0], 'vertical': 500. * units.hPa}]
-u_geo, v_geo = mpcalc.geostrophic_wind(heights, dx, dy, lat)
-print(u_geo)
-print(v_geo)
+#########################################################################
+# Unit conversion:
+
+temperature_degC = temperature[0].metpy.convert_units('degC')
+temperature_degC
+
+#########################################################################
+# Unit conversion for coordinates:
+heights_on_hPa_levels = heights.metpy.convert_coordinate_units('isobaric3', 'hPa')
+heights_on_hPa_levels['isobaric3']
+
+#########################################################################
+# Accessing just the underlying unit array:
+heights_unit_array = heights.metpy.unit_array
+heights_unit_array
 
 #########################################################################
-# Also, a limited number of calculations directly support xarray DataArrays or Datasets (they
-# can accept *and* return xarray objects). Right now, this includes
-#
-# - Derivative functions
-#     - ``first_derivative``
-#     - ``second_derivative``
-#     - ``gradient``
-#     - ``laplacian``
-# - Cross-section functions
-#     - ``cross_section_components``
-#     - ``normal_component``
-#     - ``tangential_component``
-#     - ``absolute_momentum``
-# - Smoothing functions
-#     - ``smooth_gaussian``
-#     - ``smooth_n_point``
-#     - ``smooth_window``
-#     - ``smooth_rectangular``
-#     - ``smooth_circular``
-#
-# More details can be found by looking at the documentation for the specific function of
-# interest.
+# Accessing just the underlying units:
+height_units = heights.metpy.units
+height_units
 
 #########################################################################
-# There is also the special case of the helper function, ``grid_deltas_from_dataarray``, which
-# takes a ``DataArray`` input, but returns unit arrays for use in other calculations. We could
-# rewrite the above geostrophic wind example using this helper function as follows:
+# Calculations
+# ------------
+#
+# MetPy's xarray integration extends to its calcuation suite as well. Most grid-capable
+# calculations (such as thermodynamics, kinematics, and smoothers) fully support xarray
+# ``DataArray``\s by accepting them as inputs, returning them as outputs, and automatically
+# using the attached coordinate data/metadata to determine grid arguments
+
+heights = data_parsed.metpy.parse_cf('Geopotential_height_isobaric').metpy.sel(
+    time='2017-09-05 18:00',
+    vertical=50000.
```

Comment:
Could we not use `vertical=500. * units.hPa` here?

Reply:
We definitely could, but it was a deliberate choice since unit features are not covered until the next section. There is also a parenthetical remark about that right after this code segment.

Reply:
I think you're crossed-up? There's the one like this [above](https://github.com/Unidata/MetPy/pull/1605/files#diff-ee88de57095f3c4075ffb99ef286cfe446567fdfe4cff79b4e95d9cf902144e8R100) that *is* before the units. But at this point, we used the units for selection [already](https://github.com/Unidata/MetPy/pull/1605/files#diff-ee88de57095f3c4075ffb99ef286cfe446567fdfe4cff79b4e95d9cf902144e8R219).

Reply:
Oh, yeah...I should probably have read the line numbers. Thanks for catching that!